File: Linear Queue/linearqueue.py
class Queue:

    def __init__(self,  size:int = 0, queue = []):
        self.queue = queue
        self.front = -1
        self.rear = len(queue)
        if ( size == 0 ):
            self.size = len(queue)
        elif size < 0 or size < len(queue): 
            print("size is invalid")
        else:
            self.size = size

    def is_empty(self):
        if (self.front == self.rear + 1) or (self.rear == -1):
            return True

        return False

    def is_full(self):
        if (self.rear == self.size-1):
            return True
        return False
    
    def enqueue(self, val):
        if (self.is_empty()):
            self.front+=1
            
        if (self.is_full()):
            print("Queue is full")
        else:
            self.rear+=1
            self.queue[self.rear] = val

    def dequeue(self):

        if (self.is_empty()):
            print("Queue is empty")
        else:
            self.front += 1
            print(self.queue[self.front])

    def dequeue_all(self):

        if (self.is_empty()):
            print("Queue is empty")
        else:
            while( not self.is_empty() ):
                print(self.queue[self.front])
                self.front+=1
        

File: Linear Queue/linearqueue.cpp
#include <iostream>
using namespace std;
#define size 10

class queue{

    private:
        int arr[size];
        int front, rear;

    public:
        
        queue(){
            front = -1;
            rear = -1;
        }

        bool is_empty(){
            if(front==(rear+1) || rear == -1 ){
                return true;
            }
            return false;
        }
        bool is_full(){
            if(rear == size-1){
                return true;
            }
            return false;
        }

        void enqueue(int val){
            if (is_empty()){
                front= -1; rear= -1;
                front++;
            }
            if( is_full() ){
                cout<<"Queue is full";
            }
            else{
                rear++;
                arr[rear] = val;
            }

        }

        void enqueue(int arr[], int n){

            if (is_empty()){
                front= -1; rear= -1;
                front++;
            }
            if( is_full() ){
                cout<<"Queue is full";
            }
            else{
                for ( int x=0; x < n; x++ ){
                    rear++;
                    queue::arr[rear] = arr[x];
                }
            }

        }

        void dequeue(){
            if(is_empty()) cout<<"Queue is empty";

            else front++;
        }

        void dequeue_all(){
            if(is_empty())
                cout<<"Queue is empty";

            else{
                for(int i = front; i <= rear; i++){
                    cout<<arr[i]<<"\t";
                }
            }
        }

        int peek(){
            return arr[front];
        }
        

};

int main(){

    queue q;
    int arr[5] = {1,3,432,56,7};

    q.enqueue( arr, 5);
    
    q.dequeue();
    cout<<q.peek();

    return 0;
}